### ISSUE TITLE
Update to 1.20.1

### ISSUE THREAD

**Issue by davethemodderguy** [pWn3d1337/Techguns2#301]
This thing is a total masterpiece please don't say you just gonna let it here and hit the dust for like 10 years

**Comment by Panthr75** [comment]
I am in the process of porting to 1.20.1. If anyone wants to help with textures, that would be greatly appreciated

**Comment by davethemodderguy** [comment]
I'm a bit of modeler myself. Also, if you want help with the Left to
shoot/Right Click to ADS on your mod, I'm happy to help

Vào Th 3, 20 thg 8, 2024 vào lúc 06:33 FluffyDoggo <
***@***.***> đã viết:

**Comment by Panthr75** [comment]
Once my project is able to compile, I'll set up a github for it.
![image](https://github.com/user-attachments/assets/b2855cef-b015-4111-8262-aba816068814)

Screenshots of the ammo press:
![image](https://github.com/user-attachments/assets/62bd22d8-b9e0-47a8-bbf9-c76766201bc4)

![image](https://github.com/user-attachments/assets/8ca7a3aa-116f-432e-bda8-11ce2819bbc8)

**Comment by Panthr75** [comment]
Still debating how to go about this due to the licensing. I could fork this repo, delete everything and add my changes, and make a pull request, but that's probably not the best solution.

It doesn't help that this project doesn't compile as is due to ftb-lib's maven repo getting taken down for 1.12.2

I saw that the author was working on a 1.18 fabric port, but it hasn't been updated in ~a year.

If @pWn3d1337 can help for clarity, that would be greatly appreciated.

**Comment by pWn3d1337** [comment]
I'm ok with having a 1.20 branch in this repo. I'm also willing to help out, I just don't have the time to handle the whole port by myself. 
Do you already have some progress? Because porting Techguns is an insane amount of work, don't underestimate it.

**Comment by Panthr75** [comment]
@pWn3d1337 Yes I do have some progress, as seen by that ammo press screenshots. As stated above, once I get the project in a compileable state, I'll make a draft pull request, so you can comment on the state of the code.

I've tried to retain as much of the original mods functionality as possible, but I did change
- the process on how titanium is aquired. Due to the whole ore dropping raw variant and smelting that, Titanium will drop Raw Ore and you'd process that into titanium chunks
- all machines now accept a processing time in their recipe (except for the charging station)
- ammo is more dynamic to allow for other mods to add their own guns and ammo with ammo variants

I've also set the project up with data generation. I saw that you somewhat have done this in your code but with forge 1.20 we have a separate gradle run to run data generation.

While you are here, do you have any suggestions for new textures/model changes?

Thank you for your reply.

**Comment by Panthr75** [comment]
Alright got rid of all of the compiler errors. Will you make a branch I can make a PR to?

**Comment by pWn3d1337** [comment]
https://github.com/pWn3d1337/Techguns2/tree/mc_1.20
@Panthr75

**Comment by Panthr75** [comment]
@pWn3d1337 pull request has been made

**Comment by Danmccool** [comment]
@Panthr75 Can You Try And Build The Mod I Want to Test It on 1.20.1

**Comment by Panthr75** [comment]
@Danmccool As stated in the PR, the port isn't feature complete especially with models and guns aren't fully implemented yet. I'll probably make some more pull requests in my free time to update the 1.20.1 branch

**Comment by Danmccool** [comment]
@Panthr75 Sounds Good

**Comment by augustinde** [comment]
@Panthr75  Hello, do you have any news about the 1.20.1 port?

**Comment by K0robeinik** [comment]
@Panthr75 Hello, I downloaded the 1.20.1 port and tried to play, and I fixed some duplicate registration errors and the error with the bioblob's bounding box. I'm pretty new to modding and Github, so how can I send you my changes?

**Comment by Danmccool** [comment]
@K0robeinik Do You Have The jar file and can you send it to me please

**Comment by K0robeinik** [comment]
@Danmccool Dc?
Btw It's not like the mod has any funtionality. No guns, armor or even recipes. Only the bugged ammo press.

**Comment by K0robeinik** [comment]
@Panthr75 Hi! Are you still working on the TG port/rewrite? It's been three months since the PR. If you abandoned it already, just let me know, and I'll continue working on it. If not, can I join forces with you? By the way, I'm already working on a mod for 1.20.1. You can see the repository and evaluate my skills.

**Comment by Syvelix** [comment]
Looking forward for the port

**Comment by Syvelix** [comment]
@Panthr75 how its going?

**Comment by K0robeinik** [comment]
> @Panthr75 how its going?

It appears to be abandoned. I made a PR to fix some errors, but no one replied yet.

**Comment by Syvelix** [comment]
Uh thats sad

**Comment by Syvelix** [comment]
That's still sad

**Comment by K0robeinik** [comment]
Yk what? It might be in a month, a year, or even a decade, but one day, I'll work on a mod inspired by this mod and its unfinished content. I'm not gonna copy the assets/name due to copyright, but I think it's ok if you copy just the features.

**Comment by pajamallama-sudo** [comment]
> I am in the process of porting to 1.20.1. If anyone wants to help with textures, that would be greatly appreciated

port to 1.19.2 after 1.20.1
